lib/index.js

```diff
@@ -289,8 +302,7 @@ internals.getParamsData = function (param, name, typeName) {
         meta: param.meta,
         unit: param.unit,
         type: type,
-        allowedValues: type !== 'reference' && param.valids ? internals.getExistsValues(type, param.valids) : null,
-        disallowedValues: type !== 'reference' && param.invalids ? internals.getExistsValues(type, param.invalids) : null,
+        disallowedValues: param.invalids ? internals.getExistsValues(originalType, param.invalids) : null,
```

Comment:
It seems like we're losing functionality here, maybe lacking a test on Joi.invalid(Joi.ref(...)).